lib/AST/FeatureSet.cpp

```diff
@@ -506,6 +506,26 @@ UNINTERESTING_FEATURE(Embedded)
 UNINTERESTING_FEATURE(Volatile)
 UNINTERESTING_FEATURE(SuppressedAssociatedTypes)
 
+static bool disallowFeatureSuppression(StringRef featureName, Decl *decl);
+
+static bool allBoundTypesAreCopyable(Type type, DeclContext *context) {
+  assert(type->getAnyNominal());
+  auto bgt = type->getAs<BoundGenericType>();
+  if (!bgt)
+    return false;  // nothing is bound.
+
+  for (auto argInterfaceTy : bgt->getGenericArgs()) {
+    auto argTy = context->mapTypeIntoContext(argInterfaceTy);
```

Comment:
There is no guarantee that the _generic argument_ of a BGT is going to be a valid interface type for the generic signature of the BGT.

```
struct Optional<T> {}

struct Foo<A, B: Sequence> where B.Element: Comparable {
  var x: Optional<B.Element>
}
```